functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,230 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+
+export const importChunks = functions.runWith({
+    timeoutSeconds: 300,
+    memory: '1GB'
+}).https.onRequest( async (req, res) => {
+    const auth = await google.auth.getClient({
+        scopes: ["https://www.googleapis.com/auth/spreadsheets.readonly"]
+    });
+
+    const sheets = google.sheets({ version: "v4", auth });
+    const spreadsheetId = functions.config().reporting.content.processing.spreadsheet_id;
+    let { sheetTitles, columnsIndex } = await helpers.buildSheetIndex(sheets, spreadsheetId);
+
+    let resolutions = ['OK', 'Drop', 'Duplicate', 'On hold', 'Reallot', 'Repeat', 'Derivative'];
+    let resolution, fidCheckRes; // variables to hold values of Res and fidRes when looping through the rows
+
+
+    let lastFile, // used to increment the chunks of an audiofile coming from multiple rows into a single object
+        summary = {
+            addedChunks: 0, 
+            emptyValues: 0, 
+            misplaced: 0, 
+            noResolution: 0, // rows with No `Resolution` or No `FidelityCheckResolution` values
+            overlappingChunks: [], // Chunks with overlapping times
+            modified: [] // Rows already found in the DB
+        };
+
+    
+    for (let i = 0; i < sheetTitles.length; i++) { // Loop throughout the sheets of interest
```

Comment:
Can't `forEach` be used here?

Reply:
And processing a sheet should be extracted into a separate function. I had pointed out earlier that you need to re-structure the code, but you seem to have ignored that.

Reply:
You extracted the helpers. Here I'm speaking of separating the main function logic itself.